internal/settings/providers/netcup/dnsrecord.go

```diff
@@ -0,0 +1,48 @@
+package netcup
+
+type DNSRecord struct {
+	ID           string `json:"id"`
+	DeleteRecord bool   `json:"deleterecord"`
+	Destination  string `json:"destination"`
+	Hostname     string `json:"hostname"`
+	Priority     string `json:"priority"`
+	State        string `json:"state"`
+	Type         string `json:"type"`
+}
+
+func NewDNSRecord(hostname, dnstype, destination string) *DNSRecord {
+	return &DNSRecord{
+		Destination: destination,
+		Hostname:    hostname,
+		Type:        dnstype,
+	}
+}
```

Comment:
A constructor feels unneeded, just create the struct directly without a constructor function